### PR TITLE
8293691: converting a defined BasicType value to a string should not crash the VM

### PR DESCRIPTION
**Pull request by lutkerd**
Backporting followup fix to JDK-8287491

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8293691](https://bugs.openjdk.org/browse/JDK-8293691): converting a defined BasicType value to a string should not crash the VM

### Reviewers
 * [Paul Hohensee](https://openjdk.org/census#phh) (@phohensee - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk17u-dev pull/1094/head:pull/1094` \
`$ git checkout pull/1094`

Update a local copy of the PR: \
`$ git checkout pull/1094` \
`$ git pull https://git.openjdk.org/jdk17u-dev pull/1094/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1094`

View PR using the GUI difftool: \
`$ git pr show -t 1094`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk17u-dev/pull/1094.diff">https://git.openjdk.org/jdk17u-dev/pull/1094.diff</a>

</details>

### CODE REVIEW

**Review by phohensee — COMMENTED** [review]
The patch looks like it includes more than just the original commit. Is there perhaps a predecessor backport that needs doing? Or perhaps the PR needs rebasing?

**Comment by lutkerd** [comment]
I followed the instructions in this comment: https://github.com/openjdk/jdk17u-dev/pull/1094#issuecomment-1403990903 which say to merge master. Typically when going a rebase the bot complains and says not to do a rebase for a PR in flight. I think when the bot squashes this the extra code coming in from the merge of master should drop off as being already applied, but I am not 100% sure on that.

**Review by phohensee — COMMENTED** [review]
Ok. I'll wait for the pre-submit tests to finish and approve if all is well.

**Review by phohensee — CHANGES_REQUESTED** [review]
The pre-submit Windows build failures look to be unrelated.

The change to java.c is from https://bugs.openjdk.org/browse/JDK-8286694, and the change to Awt2dLibraries.gmk from https://bugs.openjdk.org/browse/JDK-8300692. Please remove these and backport them separately.

**Comment by lutkerd** [comment]
Looks like other changes I was testing with GCC 12 also got pulled into that branch on merge. I've updated the PR to have a rebase on the master branch to clean up the extra commits that should be there.

**Comment by lutkerd** [comment]
Failing pre-submit windows failures are due to the ongoing JTReg build issue.

**Review by phohensee — APPROVED** [review]
Lgtm.

**Pull request closed without merging**
